custom_components/control4/alarm_control_panel.py

```diff
@@ -76,6 +66,24 @@
     "text": "DISPLAY_TEXT",
 }
 
+# Map Control4 states to Home Assistant states
+CONTROL4_STATE_MAPPING = {
+    CONTROL4_EXIT_DELAY_STATE: AlarmControlPanelState.ARMING,
+    CONTROL4_ENTRY_DELAY_STATE: AlarmControlPanelState.PENDING,
+    CONTROL4_ARMED_STATE: AlarmControlPanelState.ARMED_AWAY,
+    CONTROL4_DISARMED_NOT_READY_STATE: AlarmControlPanelState.DISARMED,
+    CONTROL4_DISARMED_READY_STATE: AlarmControlPanelState.DISARMED,
+}
+
+# Map Control4 armed types to Home Assistant states
+CONTROL4_ARMED_TYPE_MAPPING = {
+    DEFAULT_ALARM_AWAY_MODE: AlarmControlPanelState.ARMED_AWAY,
+    DEFAULT_ALARM_HOME_MODE: AlarmControlPanelState.ARMED_HOME,
+    DEFAULT_ALARM_NIGHT_MODE: AlarmControlPanelState.ARMED_NIGHT,
+    DEFAULT_ALARM_CUSTOM_BYPASS_MODE: AlarmControlPanelState.ARMED_CUSTOM_BYPASS,
+    DEFAULT_ALARM_VACATION_MODE: AlarmControlPanelState.ARMED_VACATION,
+}
+
```

Comment:
We cannot hardcode the ARMED_TYPE_MAPPING because each Control4 system has different names for each of the Arm states. I will revert this to use the old system of configuring the Arm state names through the HA config entry.